### PR TITLE
Feat -> UX related

### PR DESCRIPTION
**Pull request by brookmg**
+ NestedScrollView is better to handle Recycler views inside scrolling view-group
+ Hide the Fab when the user scrolls down so that the last item can be seen completely.
+ Show the app's icon and name at the bottom of the image when a given transaction is shared ( as an image ).

### CODE REVIEW

File: app/src/main/java/dev/spikeysanju/expensetracker/view/dashboard/DashboardFragment.kt [hunk 1]
```diff
@@ -183,6 +185,17 @@ class DashboardFragment :
...
+            NestedScrollView.OnScrollChangeListener { _, sX, sY, oX, oY ->
+                if (abs(sY - oY) > 10) {
+                    when {
+                        sY > oY -> btnAddTransaction.hide()
+                        oY > sY -> btnAddTransaction.show()
+                    }
+                }
+            }
+        )
+
```
Comment (Spikeysanju) [inline app/src/main/java/dev/spikeysanju/expensetracker/view/dashboard/DashboardFragment.kt:198]: Looks Good 👍

File: app/src/main/res/layout/content_transaction_details.xml [hunk 2]
```diff
@@ -193,6 +193,30 @@
...
             tools:text="Sunday, 18 Dec 2021" />
 
+        <com.google.android.material.imageview.ShapeableImageView
+            android:id="@+id/app_icon_for_share"
+            app:srcCompat="@drawable/ic_logo"
+            android:visibility="invisible"
+            app:layout_constraintEnd_toStartOf="@+id/app_name_for_share"
+            app:layout_constraintBottom_toBottomOf="parent"
+            android:layout_width="50dp"
+            android:layout_height="50dp" />
```
Comment (Spikeysanju) [inline app/src/main/res/layout/content_transaction_details.xml:203]: @brookmg Buddy shall we use `dimen_64` for both ?

File: app/src/main/res/layout/content_transaction_details.xml [hunk 3]
```diff
@@ -193,6 +193,30 @@
...
+
+        <TextView
+            android:id="@+id/app_name_for_share"
+            android:padding="16dp"
```
Comment (Spikeysanju) [inline app/src/main/res/layout/content_transaction_details.xml:207]: @brookmg Buddy shall we use `dimens_16` here

Comment (brookmg) [reply app/src/main/res/layout/content_transaction_details.xml:207]: Alright, will use references. For the above one as well.

File: app/src/main/java/dev/spikeysanju/expensetracker/view/details/TransactionDetailsFragment.kt [hunk 4]
```diff
@@ -126,19 +128,26 @@ class TransactionDetailsFragment : BaseFragment<FragmentTransactionDetailsBindin
...
+        binding.transactionDetails.appIconForShare.show().let {
+            binding.transactionDetails.appNameForShare.show().let {
+                val imageURI = binding.transactionDetails.detailView.drawToBitmap().let { bitmap ->
+                    binding.transactionDetails.appIconForShare.hide()
+                    binding.transactionDetails.appNameForShare.hide()
+                    saveBitmap(requireActivity(), bitmap)
+                } ?: run {
+                    toast("Error occurred!")
+                    return
+                }
```
Comment (Spikeysanju) [inline app/src/main/java/dev/spikeysanju/expensetracker/view/details/TransactionDetailsFragment.kt:141]: @brookmg  This is looks good. Nested scope functions will cause readability issue. Shall we try this way? 👇

Screenshot 1942-12-09 at 23 10 24

File: app/src/main/java/dev/spikeysanju/expensetracker/view/dashboard/DashboardFragment.kt (hunk 1 above)
Comment (brookmg) [reply app/src/main/java/dev/spikeysanju/expensetracker/view/dashboard/DashboardFragment.kt:198]: 👍

**Review by Spikeysanju — CHANGES_REQUESTED** [review]
LGTM!... Very minor changes needs to be done. Happy to merge the PR once requested changes are done. Have a great day!

File: app/src/main/java/dev/spikeysanju/expensetracker/view/details/TransactionDetailsFragment.kt (hunk 4 above)
Comment (brookmg) [reply app/src/main/java/dev/spikeysanju/expensetracker/view/details/TransactionDetailsFragment.kt]: Sure. This is a much better way to do it.

**Review by Spikeysanju — APPROVED** [review]
LGTM!... PR Approved 👍

**Pull request merged**